flux-tests/tests/todo/list.rs

```diff
@@ -0,0 +1,67 @@
+#[flux::sig(fn(i32{v:false}) -> T)]
+pub fn never<T>(_x: i32) -> T {
+    loop {}
+}
+
+#[derive(Debug)]
+pub enum List<T> {
+    Nil,
+    Cons(T, Box<List<T>>),
+}
+
+pub fn len<T>(l: &List<T>) -> i32 {
+    match l {
+        List::Nil => 0,
+        List::Cons(_, tl) => 1 + len(&*tl),
+    }
+}
+
+pub fn head<T>(l: &List<T>) -> &T {
+    match l {
+        List::Nil => never(0),
+        List::Cons(h, _) => h,
+    }
+}
+
+pub fn tail<T>(l: &List<T>) -> &List<T> {
+    match l {
+        List::Nil => never(0),
+        List::Cons(_, t) => &*t,
```

Comment:
Same here about deref coercion.
```suggestion
        List::Cons(_, t) => t,
```